server/src/test/java/org/elasticsearch/action/support/IndicesOptionsTests.java

```diff
@@ -134,4 +134,21 @@ public void testFromParameters() {
         assertEquals(defaultOptions.forbidClosedIndices(), updatedOptions.forbidClosedIndices());
         assertEquals(defaultOptions.ignoreAliases(), updatedOptions.ignoreAliases());
     }
+
+    public void testSimpleByteBWC() {
+        // Corresponds to byte 24 prior to rewriting IndicesOptions not to be byte-based
+        IndicesOptions indicesOptions = IndicesOptions.fromOptions(false, false, false, true, false, false, false);
+
+        logger.info("--> 1 {}", indicesOptions.toString());
+        IndicesOptions indicesOptions2 = IndicesOptions.fromByte((byte)24);
+        logger.info("--> 2 {}", indicesOptions2.toString());
+
+        assertThat(indicesOptions2.ignoreUnavailable(), equalTo(indicesOptions.ignoreUnavailable()));
+        assertThat(indicesOptions2.allowNoIndices(), equalTo(indicesOptions.allowNoIndices()));
+        assertThat(indicesOptions2.expandWildcardsOpen(), equalTo(indicesOptions.expandWildcardsOpen()));
+        assertThat(indicesOptions2.expandWildcardsClosed(), equalTo(indicesOptions.expandWildcardsClosed()));
+        assertThat(indicesOptions2.forbidClosedIndices(), equalTo(indicesOptions.forbidClosedIndices()));
+        assertThat(indicesOptions2.allowAliasesToMultipleIndices(), equalTo(indicesOptions.allowAliasesToMultipleIndices()));
+        assertThat(indicesOptions2.ignoreAliases(), equalTo(indicesOptions.ignoreAliases()));
+    }
```

Comment:
nit: shall we test the update equals/hashcode as well?

Reply:
I found that there are a whooooole lot of tests that check that (which is why I had to add them, because lots of tests failed), but I can add one if you think it's helpful, do you think I should?

Reply:
which tests do you mean? do those tests test equals/hashcode explicitly ?

Reply:
> which tests do you mean?

There are quite a few integration tests that test that the options are the same after serialization/deserialization, they ended up failing since `IndicesOptions` didn't implement equals/hashcode

> do those tests test equals/hashcode explicitly ?

They do the `assertEquals(options1, options2)` if I recall correctly.

I also changed the `testSimpleByteBWC` to use equality checks (which will use equals/hashcode)

Reply:
I think it would be nice then to test equals/hashcode separately. We can probably use EqualsHashcodeTestUtils

Reply:
Okay I added a test for this